test/unit/master/test_cluster_master.py

```diff
@@ -216,7 +216,10 @@ def test_handle_result_reported_from_slave_when_build_is_canceled(self):
         build_id = 1
         slave_url = "url"
         build = Build(BuildRequest({}))
+        self.patch('app.master.build.util')
+        build.generate_project_type()
```

Comment:
Now as the `cancel callback` is calling `project_type.kill_subprocesses` we need to generate `project_type` object.